WalletWasabi/Blockchain/Transactions/TransactionStore.cs

```diff
@@ -1,411 +1,263 @@
 using NBitcoin;
-using Nito.AsyncEx;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
 using System.IO;
 using System.Linq;
+using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
-using WalletWasabi.Blockchain.Transactions.Operations;
 using WalletWasabi.Extensions;
 using WalletWasabi.Helpers;
-using WalletWasabi.Io;
 using WalletWasabi.Logging;
-using WalletWasabi.Nito.AsyncEx;
+using WalletWasabi.Stores;
 
 namespace WalletWasabi.Blockchain.Transactions;
 
 public class TransactionStore : IAsyncDisposable
 {
-	public TransactionStore(string workFolderPath, Network network)
+	public TransactionStore(string workFolderPath, Network network, bool migrateData = true)
 	{
-		WorkFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
-		Network = network;
+		workFolderPath = Guard.NotNullOrEmptyOrWhitespace(nameof(workFolderPath), workFolderPath, trim: true);
 
-		// In Transactions.dat every line starts with the tx id, so the first character is the best for digest creation.
-		TransactionsFileManager = new IoManager(filePath: Path.Combine(WorkFolderPath, "Transactions.dat"));
+		if (workFolderPath == SqliteStorageHelper.InMemoryDatabase)
+		{
+			DataSource = SqliteStorageHelper.InMemoryDatabase;
+		}
+		else
+		{
+			IoHelpers.EnsureDirectoryExists(workFolderPath);
+			DataSource = Path.Combine(workFolderPath, "Transactions.sqlite");
+
+			// TODO: Remove. Useful for testing.
+			// if (File.Exists(dataSource))
+			// {
+			// 	 File.Delete(dataSource);
+			// }
+		}
+
+		SqliteStorage = TransactionSqliteStorage.FromFile(dataSource: DataSource, network);
+
+		if (migrateData)
+		{
```

Comment:
IMHO it's good idea to add log message that migration is being performed from old to new file.

Reply:
Why you would not want to do it normally? I would expect at least DEBUG level message with included filenames if something is moved from one location to another and some data conversion is being made. And it could help with debugging user problems if something goes wrong.

Reply:
Warning seems too harsh, it's expected operation, not something to be afraid of.

Reply:
Yes, info seems ok.